Warning during SDDM build

SDDM builds pretty cleanly. There is only one Warning:

```
[ 76%] Building CXX object src/CMakeFiles/sddm.dir/daemon/PowerManager.cpp.o                                                                
/tmp/packerbuild-1000/sddm-qt5-git/sddm-qt5-git/src/sddm/src/daemon/Greeter.cpp: In member function 'bool SDDM::Greeter::start()':
/tmp/packerbuild-1000/sddm-qt5-git/sddm-qt5-git/src/sddm/src/daemon/Greeter.cpp:86:56: warning: ignoring return value of 'int chown(const char*, __uid_t, __gid_t)', declared with attribute warn_unused_result [-Wunused-result]
             chown(qPrintable(m_socket), pw->pw_uid, -1);
                                                        ^
```

Maybe it should be fixed before they cumulate ;)

--- a/src/daemon/Greeter.cpp
+++ b/src/daemon/Greeter.cpp
@@ -82,8 +82,11 @@ namespace SDDM {
             m_process->setGid(pw->pw_gid);
 
             // take ownership of the socket so we can read/write to it
-            // -1 = don't change group
-            chown(qPrintable(m_socket), pw->pw_uid, -1);
+            if (chown(qPrintable(m_socket), pw->pw_uid, pw->pw_gid) == -1) {
+                qCritical() << "Failed to change owner of socket to user sddm.";
+                return false;
+            }
+
         }
 
         // delete process on finish

